analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpSyntaxHelper.cs

```diff
@@ -296,28 +296,43 @@ public static bool IsInExpressionTree(this SyntaxNode node, SemanticModel semant
         public static bool HasBodyOrExpressionBody(this BaseMethodDeclarationSyntax node) =>
             node?.Body != null || node?.ExpressionBody() != null;
 
-        public static SimpleNameSyntax GetIdentifier(this ExpressionSyntax expression) =>
-            expression switch
+        public static SyntaxToken? GetIdentifier(this SyntaxNode node) =>
+            node switch
```

Comment:
It's interesting that for this switch only 16 conditions out of 30 are covered. Do you know why is this the case?
Bad support of pattern matching? the fact that for constructs like `{ Alias.Identifier: var i }` we don't have a test case with null `Alias`?

Reply:
I also think that there is not a lot of added value in writing those. I was just curious if you checked already.